### PR TITLE
Display environment information

### PR DESCRIPTION
**Pull request by sschneid**

### CODE REVIEW

File: packages/dcos-integration-test/extra/test_composition.py [hunk 1]
```diff
@@ -14,6 +15,18 @@
...
+
+    cluster_environment = {
+        "docker_client_version": _docker_version('{{.Client.Version}}'),
+        "docker_server_version": _docker_version('{{.Server.Version}}'),
+        "system_platform": platform.platform(),
+        "system_platform_system": platform.system(),
+        "system_platform_release": platform.release(),
+        "system_platform_version": platform.version()
+    }
+    print(json.dumps(cluster_environment, sort_keys=True, indent=4))
```
Comment (margaret) [inline packages/dcos-integration-test/extra/test_composition.py]: looks like this file uses the `logging` module, probably want to use log.info

File: packages/dcos-integration-test/extra/test_composition.py [hunk 2]
```diff
@@ -14,6 +15,18 @@
 
 @pytest.mark.first
 def test_dcos_cluster_is_up(dcos_api_session):
+    def _docker_version(component):
+        return subprocess.check_output(['docker', 'version', '-f', component]).decode().rstrip()
```
Comment (orsenthil) [inline packages/dcos-integration-test/extra/test_composition.py]: You could provide an encoding like `.decode('utf-8','ignore')` with decode.

File: packages/dcos-integration-test/extra/test_composition.py [hunk 3]
```diff
@@ -14,6 +15,18 @@
...
+    cluster_environment = {
+        "docker_client_version": _docker_version('{{.Client.Version}}'),
+        "docker_server_version": _docker_version('{{.Server.Version}}'),
+        "system_platform": platform.platform(),
+        "system_platform_system": platform.system(),
+        "system_platform_release": platform.release(),
+        "system_platform_version": platform.version()
+    }
+    print(json.dumps(cluster_environment, sort_keys=True, indent=4))
     pass
```
Comment (orsenthil) [inline packages/dcos-integration-test/extra/test_composition.py]: Remove `pass`

File: packages/dcos-integration-test/extra/test_composition.py [hunk 4]
```diff
@@ -14,6 +15,18 @@
 
 @pytest.mark.first
 def test_dcos_cluster_is_up(dcos_api_session):
+    def _docker_version(component):
+        return subprocess.check_output(['docker', 'version', '-f', component]).decode().rstrip()
+
+    cluster_environment = {
+        "docker_client_version": _docker_version('{{.Client.Version}}'),
+        "docker_server_version": _docker_version('{{.Server.Version}}'),
```
[the remark this answers is not in the document]
Comment (BenWhitehead) [reply packages/dcos-integration-test/extra/test_composition.py]: Yeah, this drift can happen pretty often when docker-in-docker is ran.

**Comment by sschneid** [comment]
Test output sample:
```
[test_composition.test_dcos_cluster_is_up] [Test Error Output]
[2017-10-10 19:48:02,060|root|INFO]: {
    "docker_client_version": "1.12.6",
    "docker_server_version": "1.12.6",
    "system_platform": "Linux-4.7.3-coreos-r3-x86_64-Intel-R-_Xeon-R-_CPU_E5-2686_v4_@_2.30GHz-with-glibc2.3.4",
    "system_platform_release": "4.7.3-coreos-r3",
    "system_platform_system": "Linux",
    "system_platform_version": "#1 SMP Thu Feb 23 02:16:16 UTC 2017"
}
```

**Comment by sschneid** [comment]
@mesosphere-mergebot label Ready For Review

File: packages/dcos-integration-test/extra/test_composition.py [hunk 5]
```diff
@@ -14,7 +15,19 @@
 
 @pytest.mark.first
 def test_dcos_cluster_is_up(dcos_api_session):
-    pass
+    def _docker_info(component):
+        # sudo is required for non-coreOS installs
+        return subprocess.check_output(['sudo', 'docker', 'version', '-f', component]).decode('utf-8').rstrip()
```
Comment (BenWhitehead) [inline packages/dcos-integration-test/extra/test_composition.py]: Does `sudo` fail on core-os, or if the test is ran docker-in-docker where there isn't a sudo command?

Comment (sschneid) [reply packages/dcos-integration-test/extra/test_composition.py]: Yup, `sudo` [still works](https://teamcity.mesosphere.io/viewLog.html?buildId=812465&buildTypeId=DcOs_Open_Test_IntegrationTest_AwsCloudFormationSimple&tab=buildLog&_focus=127#_state=112,127) on coreOS; without `sudo` [CentOS](https://teamcity.mesosphere.io/viewLog.html?buildId=812411&buildTypeId=DcOs_Open_Test_IntegrationTest_AwsOnpremWStaticBackend&tab=buildLog&_focus=327) and [Ubuntu](https://teamcity.mesosphere.io/viewLog.html?buildId=812319&buildTypeId=DcOs_Open_Test_IntegrationTest_AzureArm&tab=buildLog&_focus=139) were failing to run `docker version` unprivileged.

Comment (sschneid) [reply packages/dcos-integration-test/extra/test_composition.py]: > if the test is ran docker-in-docker

I don't think that DinD is a scenario for the test (`test_dcos_cluster_is_up`) in question?

Comment (BenWhitehead) [reply packages/dcos-integration-test/extra/test_composition.py]: Is DinD used when these tests are run with dcos-docker?

Comment (sschneid) [reply packages/dcos-integration-test/extra/test_composition.py]: It's not run as a part of `test/docker/smoke`, but is working in `test/docker` [here](https://teamcity.mesosphere.io/viewLog.html?buildId=812463&buildTypeId=DcosIo_Dcos_DockerIntegrationTests_IntegrationTestDcosDockerPr&tab=buildLog&_focus=915#_state=163).

**Comment by sschneid** [comment]
@mesosphere-mergebot bump-ee

**Comment by sschneid** [comment]
@mesosphere-mergebot bump-ee

**Comment by sschneid** [comment]
@orsenthil @margaret could I get a (post-rebase) re-review please?

**Comment by jgehrcke** [comment]
This just landed with train 240. @sschneid is there a JIRA ticket that tracks this changed which should be updated now?

**Comment by jgehrcke** [comment]
This was not cleanly transferred to train 240. Closing manually.

**Pull request closed without merging**